src/components/Modals/CreateItemModal/CreateItemModal.tsx

```diff
@@ -2,70 +2,125 @@ import * as React from 'react'
+        item = {
+          ...addRepresentationTo,
+          data: {
+            ...addRepresentationTo.data,
+            representations: [...addRepresentationTo.data.representations],
+            replaces: [...addRepresentationTo.data.replaces],
+            hides: [...addRepresentationTo.data.hides],
+            tags: [...addRepresentationTo.data.tags]
+          },
+          contents: {
+            ...(await this.computeHashes(contents!)),
+            ...addRepresentationTo.contents
+          },
+          updatedAt: +new Date()
+        }
+
+        item.data.representations.push({
+          bodyShape: bodyShape === BodyShapeType.MALE ? [WearableBodyShape.MALE] : [WearableBodyShape.FEMALE],
           mainFile: model,
           contents: Object.keys(contents),
           overrideHides: [],
           overrideReplaces: []
         })
-      }
-
-      // add female representation
-      if (bodyShape === BodyShapeType.FEMALE || bodyShape === BodyShapeType.UNISEX) {
-        representations.push({
-          bodyShape: [WearableBodyShape.FEMALE],
-          mainFile: model,
-          contents: Object.keys(contents),
-          overrideHides: [],
-          overrideReplaces: []
-        })
-      }
-
-      const item: Item = {
-        id,
-        name,
-        thumbnail: THUMBNAIL_PATH,
-        type: ItemType.WEARABLE,
-        data: {
-          replaces: [],
-          hides: [],
-          tags: [],
-          representations
-        },
-        owner: address!,
-        metrics,
-        contents: await this.computeHashes(contents!),
-        createdAt: +new Date(),
-        updatedAt: +new Date()
+      } else {
+        // create new item
+        item = {
+          id,
+          name,
+          thumbnail: THUMBNAIL_PATH,
+          type: ItemType.WEARABLE,
+          collectionId,
+          data: {
+            replaces: [],
+            hides: [],
+            tags: [],
+            representations: []
+          },
+          owner: address!,
+          metrics,
+          contents: await this.computeHashes(contents!),
+          createdAt: +new Date(),
+          updatedAt: +new Date()
+        }
+
+        // add male representation
+        if (bodyShape === BodyShapeType.MALE || bodyShape === BodyShapeType.UNISEX) {
+          item.data.representations.push({
+            bodyShape: [WearableBodyShape.MALE],
+            mainFile: model,
+            contents: Object.keys(contents),
+            overrideHides: [],
+            overrideReplaces: []
+          })
+        }
+
+        // add female representation
+        if (bodyShape === BodyShapeType.FEMALE || bodyShape === BodyShapeType.UNISEX) {
+          item.data.representations.push({
+            bodyShape: [WearableBodyShape.FEMALE],
+            mainFile: model,
+            contents: Object.keys(contents),
+            overrideHides: [],
+            overrideReplaces: []
+          })
+        }
```

Comment:
Question:
- Is it better to maybe have this on a separate function? maybe on an utils file? I'm not certain to be honest because this Component represents the creation of a new `item`, but it seems like a bit too much logic for a component? not sure

Reply:
mm i didn't add it as a utils because i don't think we would reuse it anywhere for now, also it uses stuff from the component's state, and class methods, so if i wanted to abstract it out, i would still need to pass all that stuff as arguments, which doesn't seem ideal